### ISSUE TITLE
fetch_open_orders() appears to returns empty list for Bybit testnet 

### ISSUE THREAD

**Issue by PranavJoshi2** [ccxt/ccxt#9637]
- OS: win 7
- Programming Language version: Python 3.7.1
- CCXT version: 1.53.55

The complete sample code. Just replace it with bybit testnet api.

```
import ccxt, pprint

bybit_testnet_api_key_1 = 'XXXX_key'
bybit_testnet_api_secret_1 = 'XXXX_secret'

# ------------------------ bybit testnet exchange ---------------------------
# bybit initialization 
bybit_exchange_id = 'bybit'
exchange_class = getattr(ccxt, bybit_exchange_id)
bybit_exchange = exchange_class({
    'apiKey': bybit_testnet_api_key_1,
    'secret': bybit_testnet_api_secret_1,
})
# testnet run 
bybit_exchange.set_sandbox_mode(True)
bybit_exchange.options['adjustForTimeDifference'] = True
bybit_exchange.enableRateLimit = True
bybit_exchange.options['defaultType'] = 'future'


symbol = 'BTC/USDT'

print('\n-------------------------\nfetch_order\n')
if 1:
    # works
    all_orders = bybit_exchange.fetch_order( 'b265c203-9df1-4bc9-9db2-a510bd2be992', symbol )
    all_orders = pprint.pformat(all_orders)
    print (all_orders)
    with open('delme_fetch_order.txt', 'w') as f:
        f = f.write(str(all_orders))

print('\n-------------------------\nfetch_open_orders\n')
if 1:
    # sometimes works correctly
    # sometimes yields no results i.e. []
    # sometimes yields completely wrong result (e.g. all valid values but for a non exsistant trade)
    if (bybit_exchange.has['fetchOpenOrders']):
        print('has fetchOpenOrders')
        all_open_orders = bybit_exchange.fetch_open_orders (
            symbol, 
            since = None, 
            limit = None, 
            params = {}
            )
    all_open_orders = pprint.pformat(all_open_orders)
    print(all_open_orders)
    with open('delme_all_open_orders.txt', 'w') as f:
        f = f.write(str(all_open_orders))

print('\n-------------------------\nprivate_linear_get_position_list\n')
if 1:
    # works but i really want avoid using implicit methods
    position_list = bybit_exchange.private_linear_get_position_list(
        {
            'symbol': 'BTCUSDT', 
        })
    position_list = pprint.pformat(position_list)
    print(position_list)
    with open('delme_all_positions.txt', 'w') as f:
        f = f.write(str(position_list))
```

Output:

```
(base) E:\futuresbot>python temp3.py

-------------------------
fetch_order

{'amount': 0.001,
 'average': 32409.0,
 'clientOrderId': None,
 'cost': 32.409,
 'datetime': None,
 'fee': {'cost': 0.02430675, 'currency': 'USDT'},
 'fees': [{'cost': 0.02430675, 'currency': 'USDT'}],
 'filled': 0.001,
 'id': 'b265c203-9df1-4bc9-9db2-a510bd2be992',
 'info': {'close_on_trigger': False,
          'created_time': '2021-07-23T11:23:55Z',
          'cum_exec_fee': '0.02430675',
          'cum_exec_qty': '0.001',
          'cum_exec_value': '32.409',
          'last_exec_price': '32409',
          'order_id': 'b265c203-9df1-4bc9-9db2-a510bd2be992',
          'order_link_id': '',
          'order_status': 'Filled',
          'order_type': 'Market',
          'price': '34028.5',
          'qty': '0.001',
          'reduce_only': False,
          'side': 'Buy',
          'sl_trigger_by': 'LastPrice',
          'stop_loss': '32086.5',
          'symbol': 'BTCUSDT',
          'take_profit': '32486.5',
          'time_in_force': 'ImmediateOrCancel',
          'tp_trigger_by': 'LastPrice',
          'updated_time': '2021-07-23T11:23:55Z',
          'user_id': '246274'},
 'lastTradeTimestamp': None,
 'postOnly': False,
 'price': 34028.5,
 'remaining': 0.0,
 'side': 'buy',
 'status': 'closed',
 'stopPrice': None,
 'symbol': 'BTC/USDT',
 'timeInForce': 'IOC',
 'timestamp': None,
 'trades': None,
 'type': 'market'}

-------------------------
fetch_open_orders

has fetchOpenOrders
[]

-------------------------
private_linear_get_position_list

{'ext_code': '',
 'ext_info': '',
 'rate_limit': '120',
 'rate_limit_reset_ms': '1627045641021',
 'rate_limit_status': '119',
 'result': [{'auto_add_margin': '0',
             'bust_price': '0.5',
             'cum_realised_pnl': '-35.7037898',
             'deleverage_indicator': '3',
             'entry_price': '32409',
             'free_qty': '0.001',
             'is_isolated': False,
             'leverage': '10',
             'liq_price': '162.5',
             'occ_closing_fee': '3.8e-07',
             'position_margin': '9965.69548742',
             'position_value': '32.409',
             'realised_pnl': '0.06575387',
             'risk_id': '1',
             'side': 'Buy',
             'size': '0.001',
             'stop_loss': '32086.5',
             'symbol': 'BTCUSDT',
             'take_profit': '32486.5',
             'tp_sl_mode': 'Full',
             'trailing_stop': '0',
             'unrealised_pnl': '-0.01124',
             'user_id': '246274'},
            {'auto_add_margin': '0',
             'bust_price': '0',
             'cum_realised_pnl': '1.3992776',
             'deleverage_indicator': '0',
             'entry_price': '0',
             'free_qty': '0',
             'is_isolated': False,
             'leverage': '10',
             'liq_price': '0',
             'occ_closing_fee': '0',
             'position_margin': '0',
             'position_value': '0',
             'realised_pnl': '0',
             'risk_id': '1',
             'side': 'Sell',
             'size': '0',
             'stop_loss': '0',
             'symbol': 'BTCUSDT',
             'take_profit': '0',
             'tp_sl_mode': 'Full',
             'trailing_stop': '0',
             'unrealised_pnl': '0',
             'user_id': '246274'}],
 'ret_code': '0',
 'ret_msg': 'OK',
 'time_now': '1627045641.025401'}
```
Hope i m not missing something obvious here.

**Comment by PranavJoshi2** [comment]
this is embarrassing. 
its actually an indent issue.

i am inadvertently creating 2 variables of the same "all_open_orders" name.
Faulty code
```
print('\n-------------------------\nfetch_open_orders\n')
if 1:
    # sometimes works correctly
    # sometimes yields no results i.e. []
    # sometimes yields completely wrong result (e.g. all valid values but for a non exsistant trade)
    if (bybit_exchange.has['fetchOpenOrders']):
        print('has fetchOpenOrders')
        all_open_orders = bybit_exchange.fetch_open_orders ( # <<== wrong indent here. it should be outside the if statement.
            symbol, 
            since = None, 
            limit = None, 
            params = {}
            )
    all_open_orders = pprint.pformat(all_open_orders)
    print(all_open_orders)
    with open('delme_all_open_orders.txt', 'w') as f:
        f = f.write(str(all_open_orders))
```
Corrected code
```
print('\n-------------------------\nfetch_open_orders\n')
if 1:
    # sometimes works correctly
    # sometimes yields no results i.e. []
    # sometimes yields completely wrong result (e.g. all valid values but for a non exsistant trade)
    if (bybit_exchange.has['fetchOpenOrders']):
        print('has fetchOpenOrders')
    all_open_orders = bybit_exchange.fetch_open_orders ( 
        symbol, 
        since = None, 
        limit = None, 
        params = {}
        )
    all_open_orders = pprint.pformat(all_open_orders)
    print(all_open_orders)
    with open('delme_all_open_orders.txt', 'w') as f:
        f = f.write(str(all_open_orders))
```
this one seems to work fine.

Regrading ->     # sometimes yields completely wrong result (e.g. all valid values but for a non exsistant trade)
it 'seemed' to have happen only once and i could not reproduce that error again.
even that must have been some kind of oversight. 
Although i think it wont be needed but in case if i could reproduce such error, i will open a new issue.

perhaps i was overworking a few days. took a break for a day and i can see the code much clearly now.. lol.
my bad. apology.
you can please close this issue.

**Comment by PranavJoshi2** [comment]
and even the corrected code is not working today.
todays (partial) output with the corrected code:

```
-------------------------
fetch_open_orders

has fetchOpenOrders
[]
-------------------------
private_linear_get_position_list

{'ext_code': '',
 'ext_info': '',
 'rate_limit': '120',
 'rate_limit_reset_ms': '1627178159227',
 'rate_limit_status': '119',
 'result': [{'auto_add_margin': '0',
             'bust_price': '0.5',
             'cum_realised_pnl': '-35.88060481',
             'deleverage_indicator': '2',
             'entry_price': '33946',
             'free_qty': '0.001',
             'is_isolated': False,
             'leverage': '10',
             'liq_price': '170',
             'occ_closing_fee': '3.8e-07',
             'position_margin': '9965.51867241',
             'position_value': '33.946',
             'realised_pnl': '-0.0254595',
             'risk_id': '1',
             'side': 'Buy',
             'size': '0.001',
             'stop_loss': '33788.5',
             'symbol': 'BTCUSDT',
             'take_profit': '34309',
             'tp_sl_mode': 'Full',
             'trailing_stop': '0',
             'unrealised_pnl': '-0.05068',
             'user_id': '246274'},
            {'auto_add_margin': '0',
             'bust_price': '0',
             'cum_realised_pnl': '1.3992776',
             'deleverage_indicator': '0',
             'entry_price': '0',
             'free_qty': '0',
             'is_isolated': False,
             'leverage': '10',
             'liq_price': '0',
             'occ_closing_fee': '0',
             'position_margin': '0',
             'position_value': '0',
             'realised_pnl': '0',
             'risk_id': '1',
             'side': 'Sell',
             'size': '0',
             'stop_loss': '0',
             'symbol': 'BTCUSDT',
             'take_profit': '0',
             'tp_sl_mode': 'Full',
             'trailing_stop': '0',
             'unrealised_pnl': '0',
             'user_id': '246274'}],
 'ret_code': '0',
 'ret_msg': 'OK',
 'time_now': '1627178159.231560'}
```
fetch_open_orders again seem to return empty list.

**Comment by kroitor** [comment]
@PranavJoshi2 when you fetch open orders – it will return just the open orders. When you open an order and it gets filled – your position is opened, and your open order is closed (not open anymore).

**Comment by kroitor** [comment]
You can verify whether `fetch_open_orders` returns the open orders to you, by checking those open orders on the Bybit website.

**Comment by PranavJoshi2** [comment]
ok, i changed the "private_linear_get_position_list( )" to "private_linear_get_order_search( )"

current price: around 34269.5
manually opened a limit sell at 35271 with tp & sl

Bybit testnet GUI: https://pasteboard.co/KcO6kgD.bmp
as you can see there is only 1 Active order and 0 positions

On run, the program gives following output:

```
-------------------------
fetch_open_orders

has fetchOpenOrders
[]

-------------------------
private_linear_get_order_search

{'ext_code': '',
 'ext_info': '',
 'rate_limit': '600',
 'rate_limit_reset_ms': '1627217403869',
 'rate_limit_status': '598',
 'result': [{'close_on_trigger': False,
             'created_time': '2021-07-25T12:49:55Z',
             'cum_exec_fee': '0',
             'cum_exec_qty': '0',
             'cum_exec_value': '0',
             'last_exec_price': '0',
             'order_id': 'e9abc0bb-bda7-4010-9875-713821bf4da3',
             'order_link_id': '',
             'order_status': 'New',
             'order_type': 'Limit',
             'price': '35271',
             'qty': '0.001',
             'reduce_only': False,
             'side': 'Sell',
             'sl_trigger_by': 'LastPrice',
             'stop_loss': '35450.5',
             'symbol': 'BTCUSDT',
             'take_profit': '34913',
             'time_in_force': 'GoodTillCancel',
             'tp_trigger_by': 'LastPrice',
             'updated_time': '2021-07-25T12:49:55Z',
             'user_id': '246274'}],
 'ret_code': '0',
 'ret_msg': 'OK',
 'time_now': '1627217403.872996'}
```

5-10 mins later
the correct output.. same program no change just wait.

```
-------------------------
fetch_open_orders

has fetchOpenOrders
[{'amount': 0.001,
  'average': None,
  'clientOrderId': None,
  'cost': 0.0,
  'datetime': None,
  'fee': {'cost': 0.0, 'currency': 'USDT'},
  'fees': [{'cost': 0.0, 'currency': 'USDT'}],
  'filled': 0.0,
  'id': 'e9abc0bb-bda7-4010-9875-713821bf4da3',
  'info': {'close_on_trigger': False,
           'created_time': '2021-07-25T12:49:55Z',
           'cum_exec_fee': '0',
           'cum_exec_qty': '0',
           'cum_exec_value': '0',
           'last_exec_price': '0',
           'order_id': 'e9abc0bb-bda7-4010-9875-713821bf4da3',
           'order_link_id': '',
           'order_status': 'New',
           'order_type': 'Limit',
           'price': '35271',
           'qty': '0.001',
           'reduce_only': False,
           'side': 'Sell',
           'sl_trigger_by': 'LastPrice',
           'stop_loss': '35450.5',
           'symbol': 'BTCUSDT',
           'take_profit': '34913',
           'time_in_force': 'GoodTillCancel',
           'tp_trigger_by': 'LastPrice',
           'updated_time': '2021-07-25T12:49:55Z',
           'user_id': '246274'},
  'lastTradeTimestamp': None,
  'postOnly': False,
  'price': 35271.0,
  'remaining': 0.001,
  'side': 'sell',
  'status': 'open',
  'stopPrice': None,
  'symbol': 'BTC/USDT',
  'timeInForce': 'GTC',
  'timestamp': None,
  'trades': None,
  'type': 'limit'}]

-------------------------
private_linear_get_order_search

{'ext_code': '',
 'ext_info': '',
 'rate_limit': '600',
 'rate_limit_reset_ms': '1627217892729',
 'rate_limit_status': '598',
 'result': [{'close_on_trigger': False,
             'created_time': '2021-07-25T12:49:55Z',
             'cum_exec_fee': '0',
             'cum_exec_qty': '0',
             'cum_exec_value': '0',
             'last_exec_price': '0',
             'order_id': 'e9abc0bb-bda7-4010-9875-713821bf4da3',
             'order_link_id': '',
             'order_status': 'New',
             'order_type': 'Limit',
             'price': '35271',
             'qty': '0.001',
             'reduce_only': False,
             'side': 'Sell',
             'sl_trigger_by': 'LastPrice',
             'stop_loss': '35450.5',
             'symbol': 'BTCUSDT',
             'take_profit': '34913',
             'time_in_force': 'GoodTillCancel',
             'tp_trigger_by': 'LastPrice',
             'updated_time': '2021-07-25T12:49:55Z',
             'user_id': '246274'}],
 'ret_code': '0',
 'ret_msg': 'OK',
 'time_now': '1627217892.732672'}
```


came across something:
source: https://bybit-exchange.github.io/docs/linear/?python#t-getactive

**Get my active order list.**
Because order creation/cancellation is asynchronous, there can be a data delay in this endpoint. You can get real-time order info with the Query Active Order (real-time) endpoint.

**Comment by kroitor** [comment]
@PranavJoshi2 i don't see anything wrong with the above output. Can you elaborate on the problem more specifically? What is wrong? What do you expect? What do you actually get instead?

**Comment by PranavJoshi2** [comment]
i had written in the comment (in the python code) in my first post,
the code appeared to show 3 behaviors when run at different times under same conditions.
    # sometimes works correctly
    # sometimes yields no results i.e. []
    # sometimes yields completely wrong result (e.g. all valid values but for a non exsistant trade)

in my last post,
1. i entered a manual limit order (which has not become a position yet)
2. ran the program and it gave [] output
3. waited for some time and ran it again, and it gave out the desired output
in other words,
    # sometimes works correctly
    # sometimes yields no results i.e. []

```
print('\n-------------------------\nfetch_open_orders\n')
if 1:
    # sometimes works correctly
    # sometimes yields no results i.e. []
    # sometimes yields completely wrong result (e.g. all valid values but for a non exsistant trade)
    if (bybit_exchange.has['fetchOpenOrders']):
        print('has fetchOpenOrders')
        all_open_orders = bybit_exchange.fetch_open_orders (
            symbol, 
            since = None, 
            limit = None, 
            params = {}
            )
    all_open_orders = pprint.pformat(all_open_orders)
    print(all_open_orders)
    with open('delme_all_open_orders.txt', 'w') as f:
        f = f.write(str(all_open_orders))
```
there are 2 outputs at different times.

**Comment by PranavJoshi2** [comment]
may be just make do with what is and move ahead. 
there is always some workaround.
i was a software professional myself and i have a hunch that its probably not an issue of ccxt.
might as well close the issue sir.
if you none-the-less need any more input from me on this please do let me know.
-tnx
